src/Domain/Block.hpp

```diff
@@ -0,0 +1,89 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class template Block.
+
+#pragma once
+
+#include <iosfwd>
+#include <memory>
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+
+#include "Domain/BlockNeighbor.hpp"
+#include "Domain/Direction.hpp"
+#include "Domain/EmbeddingMaps/EmbeddingMap.hpp"
+
+/// \ingroup ComputationalDomain
+/// A Block<VolumeDim> is a region of a VolumeDim-dimensional computational
+/// domain that defines the root node of a tree which is used to construct the
+/// Elements that cover the computational domain.
```

Comment:
Unless our use of Blocks has changed in the past year, the elements that live in a Block's tree can only cover a _region_ of the computational domain.